pkg/kv/mem/store.go

```diff
@@ -0,0 +1,169 @@
+package mem
+
+import (
+	"bytes"
+	"context"
+	"sort"
+	"sync"
+
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type Driver struct{}
+
+type Store struct {
+	m    map[string][]byte
+	keys []string
+	mu   sync.RWMutex
+}
+
+type Entries struct {
+	entry *kv.Entry
+	err   error
+	start []byte
+	first bool
```

Comment:
I might be missing something, but I didn't find a real usage for 'first'

Reply:
good catch - at first I was planning to use it to select the first item by search for >= at first and later > based on previous key. modified the implementation to get the next key during 'Next' so flag is redundant.